core-api/src/main/java/org/silverpeas/core/admin/user/model/SilverpeasRole.java

```diff
@@ -35,17 +33,12 @@
 import java.util.EnumSet;
 import java.util.Set;
 
-import static org.silverpeas.core.admin.service.OrganizationControllerProvider.getOrganisationController;
-
 public enum SilverpeasRole {
   admin, Manager, publisher, writer, privilegedUser, user, reader, supervisor;
 
   // Unfortunately, several codes of role can define the same role nature in Silverpeas ...
   public static EnumSet<SilverpeasRole> READER_ROLES = EnumSet.of(user, reader);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make READER_ROLES a static final constant or non-public and provide accessors if needed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this "public static READER_ROLES" field final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1444)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The type of the "READER_ROLES" object should be an interface such as "Set" rather than the implementation "EnumSet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1319)